### ISSUE TITLE
Parsley >2.3.2 fails to minify correctly using Rhino

### ISSUE THREAD

**Issue by btburton42** [guillaumepotier/Parsley.js#1070]
We are using a CMS (Adobe CQ) that utilizes Mozilla's Rhino tool for JS minification and Parsley `2.3.2` is failing at various points in the script to minify properly. Is this something that's been observed or to be expected?

*using Rhino 1.7R4*

**Comment by marcandre** [comment]
No! Very odd?
A minifier should be able to minify any script, no?
Our output is made by `babel`, so it's even stranger. I'd file a bug with them...

**Comment by marcandre** [comment]
Do you have an example of code that minimizes incorrectly?

**Comment by btburton42** [comment]
Yes, very odd indeed. When we remove 2.3.4 out of our global.min.js build it will compile, otherwise it fails. I can give you a hint that it does not like this block

```
  var requirementConverters = {
    string: function string(_string) {
      return _string;
    },
    integer: function integer(string) {
      if (isNaN(string)) throw 'Requirement is not an integer: "' + string + '"';
      return parseInt(string, 10);
    },
    number: function number(string) {
      if (isNaN(string)) throw 'Requirement is not a number: "' + string + '"';
      return parseFloat(string);
    },
    reference: function reference(string) {
      // Unused for now
      var result = $(string);
      if (result.length === 0) throw 'No such reference: "' + string + '"';
      return result;
    },
    boolean: function boolean(string) {
      return string !== 'false';
    },
    object: function object(string) {
      return ParsleyUtils__default.deserializeValue(string);
    },
    regexp: function regexp(_regexp) {
      var flags = '';

      // Test if RegExp is literal, if not, nothing to be done, otherwise, we need to isolate flags and pattern
      if (/^\/.*\/(?:[gimy]*)$/.test(_regexp)) {
        // Replace the regexp literal string with the first match group: ([gimy]*)
        // If no flag is present, this will be a blank string
        flags = _regexp.replace(/.*\/([gimy]*)$/, '$1');
        // Again, replace the regexp literal string with the first match group:
        // everything excluding the opening and closing slashes and the flags
        _regexp = _regexp.replace(new RegExp('^/(.*?)/' + flags + '$'), '$1');
      } else {
        // Anchor regexp:
        _regexp = '^' + _regexp + '$';
      }
      return new RegExp(_regexp, flags);
    }
  };
```

**Comment by marcandre** [comment]
Is it possible for you to check if changing the line with `boolean` in it fixes it?
From
    boolean: function boolean(string) {
to
    'boolean': function _boolean(string) {

It looks like `boolean` was a [reserved keyword in EcmaScript 3](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Lexical_grammar#Reserved_keywords_as_of_ECMAScript_6), so maybe that's what the issue is? I don't really see why rhino would choke on it, but `babel` should not output that code either.

**Comment by marcandre** [comment]
Otherwise maybe there's some conflict with `string`, so you could try replacing

    string: function string(_string) {

with 

    string: function __string(_string) {

BTW, the names of the functions are not actually used here. `babel` creates those to ease debugging I think.

**Comment by halovanic** [comment]
I am not 100% sure this is the same issue originally reported, but I also have minification failures in the most recent versions of Parsley with the ASP.NET Web Optimization framework.  However, my error message points to the strict mode being violated for the undefined variable 'value', and this function in multiple.js:

```javascript
  getValue: function () {
    // Value could be overriden in DOM
    if ('function' === typeof this.options.value)
      value = this.options.value(this);
```

Adding `var value;` here makes my minifier happy again.

**Comment by marcandre** [comment]
@halovanic maybe not the same issue, but that's an actual bug, thanks for the report. Fixed & released.

Closing the initial report due to lack of feedback.

**Comment by marcandre** [comment]
Looks like it was only due to the 'boolean' keyword. Fixed and released as 2.6.5
